packages/debug/src/browser/debug-configuration-manager.ts

```diff
@@ -527,7 +529,12 @@ export class DebugConfigurationManager {
             return;
         }
 
-        this.recentDynamicOptionsTracker = options;
+        // ensure options name is reflected from old configurations data
+        const dynamicOptions = options.map(option => {
+            option.name = option.name ?? option.configuration.name;
+            return option;
+        }).filter(DebugSessionOptions.isDynamic);
```

Comment:
is it really necessary to filter for `isDynamic` ?
non dynamic configs should not be available under the received argument:
`options?: DynamicDebugConfigurationSessionOptions[]`, right?

Reply:
You are absolutely correct that it should not be necessary. I added this guard because you had the case where there was a regular configuration in the 'recentDynamicOptions'. I could not reproduce the case but during runtime we already have a type guard before updating recently used dynamic options so I reasoned that it probably was caused during loading some old data. Since the recent options are limited to 3, I thought the extra safety outweighs any potential performance cost. What do you think?

Reply:
I am leaning towards removing it, as the case may have been caused while switching among different versions of this change.
  If we ever see it again, it's not too bad for the user to ignore it or replace it by making a new selection, and then we could try to fix the actual source of the problem.